evap/results/tools.py

```diff
@@ -41,28 +41,43 @@ def __init__(self, visible_by_contribution: Iterable[UserProfile], visible_by_de
         self.visible_by_delegation_count = visible_by_delegation_count
 
 
+def create_rating_result(question, answer_counters, additional_text_result=None):
+    if answer_counters is None:
+        return RatingResult(question, answer_counters, additional_text_result)
+    if any(counter.count != 0 for counter in answer_counters):
+        return AnsweredRatingResult(question, answer_counters, additional_text_result)
+    return PublishedRatingResult(question, answer_counters, additional_text_result)
+
+
 class RatingResult:
-    def __init__(self, question, answer_counters, additional_text_result=None):
+    @classmethod
+    def is_published(cls, rating_result) -> TypeGuard["PublishedRatingResult"]:
+        return isinstance(rating_result, PublishedRatingResult)
+
+    @classmethod
+    def has_answers(cls, rating_result) -> TypeGuard["AnsweredRatingResult"]:
+        return isinstance(rating_result, AnsweredRatingResult)
+
+    def __init__(self, question, _, additional_text_result=None):
         assert question.is_rating_question
         self.question = discard_cached_related_objects(copy(question))
         self.additional_text_result = additional_text_result
 
-        if answer_counters is not None:
-            counts = OrderedDict((value, 0) for value in self.choices.values if value != NO_ANSWER)
-            for answer_counter in answer_counters:
-                counts[answer_counter.answer] = answer_counter.count
-            self.counts = tuple(counts.values())
-        else:
-            self.counts = None
-
     @property
     def choices(self):
         return CHOICES[self.question.type]
 
+
+class PublishedRatingResult(RatingResult):
```

Comment:
The class names are not very intuitive to me. Can we find a short comment explaining these names that we can add here? I think the last state was "This is what the code currently has, we don't really have a good argument why" :sweat_smile: